backend/infrahub/core/models.py

```diff
@@ -50,15 +52,17 @@ class SchemaBranchHash(BaseModel):
     nodes: dict[str, str] = Field(default_factory=dict)
     generics: dict[str, str] = Field(default_factory=dict)
 
-    def compare(self, other: SchemaBranchHash) -> Optional[SchemaBranchDiff]:
+    def compare(self, other: SchemaBranchHash) -> SchemaBranchDiff | None:
         if other.main == self.main:
             return None
 
         return SchemaBranchDiff(
             nodes=[key for key, value in other.nodes.items() if key not in self.nodes or self.nodes[key] != value],
+            removed_nodes=[key for key in self.nodes if key not in other.nodes],
             generics=[
                 key for key, value in other.generics.items() if key not in self.generics or self.generics[key] != value
             ],
+            removed_generics=[key for key in self.generics if key not in other.generics],
         )
```

Comment:
it might be cleaner if these used sets instead of lists for this logic, but I understand if it is out of scope for this small change

Reply:
You mean all of the objects in `SchemaBranchDiff`? If so perhaps we can do that once it's brought back into `develop`?